src/types/seed_phrase.ts

```diff
@@ -0,0 +1,35 @@
+import { Equatable } from './equatable';
+
+const seedPhraseRegex = /^(\b[a-zA-Z]+\b(\s+(?=\b[a-zA-Z]+\b)|$)){12}$/;
+
+export class SeedPhrase implements Equatable<SeedPhrase> {
+	constructor(private readonly seedPhrase: string) {
+		if (!this.seedPhrase.match(seedPhraseRegex)) {
+			throw new Error(`'${this.seedPhrase}' is not a valid 12 word seed phrase!`);
+		}
+	}
+
+	[Symbol.toPrimitive](hint?: string): string {
+		if (hint === 'number') {
+			throw new Error('Seed phrase cannot be interpreted as a number!');
+		}
+
+		return this.toString();
+	}
+
+	toString(): string {
+		return this.seedPhrase;
+	}
+
+	valueOf(): string {
+		return this.seedPhrase;
+	}
+
+	toJSON(): string {
```

Comment:
Is this function in use? If so, please add a docblock giving some context

Reply:
This method is used by `JSON.stringify` internally, that's why there's no references to it in the codebase